homeassistant/components/light/__init__.py

```diff
@@ -1209,7 +1235,7 @@ def state_attributes(self) -> dict[str, Any] | None:
                 data[ATTR_BRIGHTNESS] = self.brightness
             else:
                 data[ATTR_BRIGHTNESS] = None
-        elif supported_features_value & SUPPORT_BRIGHTNESS:
+        elif supported_features_value & _DEPRECATED_SUPPORT_BRIGHTNESS.value:
```

Comment:
Isn't this going to trigger the deprecation warning even if it's not used by the subclass?

Reply:
No it shouldn't.
It only gets triggered when it is accessed externally via the original name.